File: Array-String/HashTable/CPP/number_of_good_pairs.cpp
#include <vector>
using namespace std;

class Solution {
    public:
        int numIdenticalPairs(vector<int>& nums) {
            int ans = 0, n = nums.size();
            for (int i = 0; i < n; i++) {
                for (int j = i+1; j < n; j++) {
                    if (nums[i] == nums[j]) ans++;
                }
            }
            return ans;
        }
    };

File: Array-String/HashTable/Python/number_of_good_pairs.py
from collections import Counter

class Solution:
    def numIdenticalPairs(self, nums: list[int]) -> int:
        ans = 0
        n = len(nums)
        for i in range(n):
            for j in range(i+1, n):
                if nums[i] == nums[j]: ans += 1
        return ans

    # shorter
    def numIdenticalPairs(self, nums: list[int]) -> int:
        count = Counter(nums)
        return sum(v * (v - 1) // 2 for v in count.values())